lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2297,84 +2302,210 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
+      int64_t size;
+      if (matchPattern(en.value(), m_TorchConstantInt(&size)))
+        outputShape[outputDim] = size;
+    }
+
+    SmallVector<int64_t> collapsedShape =
+        isCollapse ? outputShape : llvm::to_vector(inputShape);
+    SmallVector<int64_t> expandedShape =
+        isCollapse ? llvm::to_vector(inputShape) : outputShape;
+
+    // The while loop does the following:
+    // 1. Fill in the reassociation indices for dimensions that are expanded.
+    // Check the interval dimensions between two unchanged dims in the
+    // collapsedShape. If the interval is size 1, associate all the dims
+    // in the expandedShape shape until the next unchanged dim. If the interval
+    // is larger than size 1, figure out the associations with assumptions that
+    // dynamic dimensions are not splitted.
+    // 2. Set collapsedShape and expandedShape following the requirements by
+    // tensor.expand_shape verification code:
+    //    a. As long as one or more of the related dimensions in the expanded
+    //    shape is dynamic the collapsed dimension is dynamic.
+    //    b. If all of the related dimensions are static, the collapsed
+    // dimension must be static. In other words, if a collapsed dimension is
+    // dynamic, at least one of the related dimensions need to be dynamic.
+    int64_t collapsedDim = 0, expandedDim = 0;
+    while (collapsedDim < collapsedRank && expandedDim < expandedRank) {
+      // Not empty means the associations has been filled in and the dimension
+      // is unchanged.
+      if (!reassociation[collapsedDim].empty()) {
+        if (expandedDim != reassociation[collapsedDim][0])
+          return op.emitOpError("Unsupported: expanded dims are off from the "
+                                "expected dim got from reassociation");
+        collapsedDim++;
+        expandedDim++;
+        continue;
+      }
+
+      // Collect the dims that are collapsed until hitting the next dim that's
+      // unchanged.
+      SmallVector<int64_t> collpasedDims;
+      while (collapsedDim < collapsedRank &&
+             reassociation[collapsedDim].empty()) {
+        collpasedDims.push_back(collapsedDim);
+        collapsedDim++;
+      }
+      // the next reassociation is for a dim that's unchanged.
+      int64_t expandedDimNext = collapsedDim != collapsedRank
+                                    ? reassociation[collapsedDim][0]
+                                    : expandedRank;
+      if (collpasedDims.size() == 1) {
+        int64_t collapsedDimSize = 1;
+        int64_t collapsedDim = collpasedDims[0];
+        for (auto i : llvm::seq<int64_t>(expandedDim, expandedDimNext)) {
+          reassociation[collapsedDim].push_back(i);
+          if (collapsedDimSize == kUnknownSize)
+            continue;
+
+          int64_t expandedDimSize = expandedShape[i];
+          if (expandedDimSize == kUnknownSize) {
+            collapsedDimSize = kUnknownSize;
+            continue;
+          }
```

Comment:
Would it be more efficient by using bool comparison? Thoughts?
bool isUnknownSize = false;
....
if(expandedDimSize == kUnknownSize) isUnknownSize = true;

Reply:
I don't follow why introducing an extra bool variable would make this more efficient.  Could you explain more?

Reply:
I was thinking about bool comparison vs int comparison, which one is faster? particularly when they are executed many times. I read about it a little bit and it seems like the difference is negligible.

Reply:
yea, I think performance optimization at this level is better left for the compiler to decide.